src/Doctrine/Instantiator/Instantiator.php

```diff
@@ -144,27 +142,25 @@ private function checkIfUnSerializationIsSupported(ReflectionClass $reflectionCl
             );
         });
 
-        $this->attemptInstantiationViaUnSerialization($reflectionClass, $serializedString);
-
-        restore_error_handler();
+        try {
```

Comment:
Please run the benchmarks before/after the patch as well: adding more `try` blocks has an impact here